libs/alcotest_ext/Alcotest_ext.mli

```diff
@@ -0,0 +1,141 @@
+
+   Alcotest API:
+   https://mirage.github.io/alcotest/alcotest/Alcotest/index.html
+*)
+
+type output = Stdout | Stderr | Merged_stdout_stderr | Separate_stdout_stderr
+
+(*
+   A test is a name and a function that raises exceptions to signal
+   test failure.
+
+   There are two main recommended ways of writing a test:
+
+   1. With 'assert false':
+
+   Each test may use 'assert false' to indicate that the test doesn't pass.
+   This is the simplest way of failing while also showing the location
+   of the failure. When using 'assert false', you should print the expected
+   value and actual value if applicable.
+
+   2. With 'Alcotest.(check ...)':
+
+   This is a little nicer because the error messages prints something like
+   "Expecting 'foo', got 'bar'". However, this makes tests more complicated
+   to write. If the test already prints the expected value and the actual
+   value as its output, it's just easier to fail with 'assert false'.
+
+   In any case, Alcotest will capture the output (stdout, stderr) of each
+   test and put it in its own file so we can consult it later. Don't
+   hesitate to log a lot during the execution of the test.
+*)
+type 'a t = private {
+  category : string list;
+  name : string;
+  func : unit -> 'a;
+  (* Options *)
+  speed_level : Alcotest.speed_level;
+  check_output : output option;
+  (* Automatically determined *)
+  id : string;
+}
+
+type test = unit t
+
+(* Legacy type that doesn't support options *)
+type simple_test = string * (unit -> unit)
+
+(* Lwt.t promises transpiled to JS via jsoo must have their Lwt.t nature
+   hoisted all the way to the top level, so we can run them properly on the
+   JS runtime.
+   When running such tests in JS, we need our tests to also return promises.
+*)
+type lwt_test = unit Lwt.t t
+
+(*
+   Create a test to appear in a test suite.
+*)
+val create :
+  ?check_output:output ->
+  ?speed_level:Alcotest.speed_level ->
+  string ->
+  (unit -> 'a) ->
+  'a t
+
+(* Convert legacy optionless format to new format *)
+val simple_test : string * (unit -> 'a) -> 'a t
+val simple_tests : (string * (unit -> 'a)) list -> 'a t list
+
+(* Register a test. The test gets added to the global list of tests.
+   This is meant to declare inline tests as follows:
+
+     let () = Alcotest_ext.test "foo" (fun () ->
+       (* test body raising exceptions to signal failure *)
+       ...
+     )
+*)
+val test :
+  ?check_output:output ->
+  ?speed_level:Alcotest.speed_level ->
+  string ->
+  (unit -> unit) ->
+  unit
+
+val test_lwt :
+  ?check_output:output ->
+  ?speed_level:Alcotest.speed_level ->
+  string ->
+  (unit -> unit Lwt.t) ->
+  unit
+
+(* Get the list of registered tests. *)
+val get_registered_tests : unit -> test list
+val get_registered_lwt_tests : unit -> lwt_test list
+
+(*
+   Usage:
+
+     pack_tests_pro "Suite Name" [test1; test2]
+     pack_suites "Suite Name" [suite1; suite2; suite3]
```

Comment:
maybe say that you can nest them, and that the name gets inserted in the category field of the test.

Reply:
This is another thing we've doing on top of Alcotest (I thought I posted a feature request for this on https://github.com/mirage/alcotest/issues but I can't find it). The nesting is purely visual i.e. we `["food"; "fruit"; "banana"]` becomes the test name `food > fruit > banana` and we can filter tests using the substring `od > fru`. It's weird but it doesn't hurt.